actors/miner/tests/change_worker_address_test.rs

```diff
@@ -0,0 +1,278 @@
+fn successfully_resolve_and_change_only_control_addresses() {
+    let (h, mut rt) = setup();
+
+    let (control_address_1, control_address_2) = (Address::new_id(555), Address::new_id(556));
+    let control_address_2_non_id = new_bls_addr(42);
+    rt.add_id_address(control_address_2_non_id, control_address_2);
+    rt.set_address_actor_type(control_address_1, *ACCOUNT_ACTOR_CODE_ID);
+    rt.set_address_actor_type(control_address_2, *ACCOUNT_ACTOR_CODE_ID);
+
+    h.change_worker_address(&mut rt, h.worker, vec![control_address_1, control_address_2_non_id])
+        .unwrap();
+
+    // assert there is no worker change request and worker key is unchanged
+    let info = h.get_info(&rt);
+    assert_eq!(h.worker, info.worker);
+    assert!(info.pending_worker_key.is_none());
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn successfully_change_both_worker_and_control_addresses() {
+    let (h, mut rt) = setup();
+
+    let new_worker = Address::new_id(999);
+    let (control_address_1, control_address_2) = (Address::new_id(5001), Address::new_id(5002));
+    rt.set_address_actor_type(control_address_1, *ACCOUNT_ACTOR_CODE_ID);
+    rt.set_address_actor_type(control_address_2, *ACCOUNT_ACTOR_CODE_ID);
+
+    let current_epoch = 5;
+    rt.set_epoch(current_epoch);
+    let effective_epoch = current_epoch + rt.policy().worker_key_change_delay;
+    h.change_worker_address(&mut rt, new_worker, vec![control_address_1, control_address_2])
+        .unwrap();
+
+    // set current epoch and update worker key
+    rt.set_epoch(effective_epoch);
+    h.confirm_update_worker_key(&mut rt).unwrap();
+
+    // assert both worker and control addresses have changed
+    let info = h.get_info(&rt);
+    assert_eq!(info.control_addresses, vec![control_address_1, control_address_2]);
+    assert_eq!(info.worker, new_worker);
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn successfully_clear_all_control_addresses() {
+    let (h, mut rt) = setup();
+
+    h.change_worker_address(&mut rt, h.worker, Vec::new()).unwrap();
+
+    // assert control addresses are cleared
+    let info = h.get_info(&rt);
+    assert!(info.control_addresses.is_empty());
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn fails_if_control_addresses_length_exceeds_maximum_limit() {
+    let (h, mut rt) = setup();
+
+    let control_addresses =
+        (0..=rt.policy().max_control_addresses as u64).map(Address::new_id).collect_vec();
+    let result = h.change_worker_address(&mut rt, h.worker, control_addresses);
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        "control addresses length",
+        result,
+    );
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn fails_if_unable_to_resolve_control_address() {
+    let (h, mut rt) = setup();
+
+    let control_address = new_bls_addr(42);
+    let result = h.change_worker_address(&mut rt, h.worker, vec![control_address]);
+    expect_abort(ExitCode::USR_ILLEGAL_ARGUMENT, result);
+
```

Comment:
rt.verify()

Reply:
already called in `change_worker_address` in case of success, reset in case of a fiasco.